### PR TITLE
Fix README.md to contain summary of the API and where to find code.

### PR DESCRIPTION
**Pull request by tcnghia**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,41 @@
...
-and [DEVELOPMENT.md](./DEVELOPMENT.md).
+In addition to these, we previously added Timeout and Retry settings but no
+longer used them and may deprecate these parts in future versions.
+
+Check out:
+* pkg/apis/networking/v1alpha1/ingress_types.go for more information about the
+  KIngress API spec.
+* pkg/apis/networking/v1alpha1/ingress_validations.go for more information about
+  the validation logic for KIngress API spec.
+* test/conformance/ingress/README.md for the conformance tests and how to run
```
Comment (mattmoor) [inline README.md:27]: may as well make all file paths links.

File: README.md [hunk 2]
```diff
@@ -1,14 +1,41 @@
-# Knative Sample Controller
+# knative/networking
 
-[![GoDoc](https://godoc.org/knative.dev/sample-controller?status.svg)](https://godoc.org/knative.dev/sample-controller)
-[![Go Report Card](https://goreportcard.com/badge/knative/sample-controller)](https://goreportcard.com/report/knative/sample-controller)
+This repository contains the Knative Ingress and Certificate CRDs, as well as
+their conformance tests. These are our extension points to plugin different
+Ingress plugins (Ambassador, Contour, Gloo, Istio, Kong and Kourier), as well as
```
Comment (mattmoor) [inline README.md:5]: perhaps redirect to docs for the latest list of integrations?

File: README.md (hunk 1 above)
Comment (tcnghia) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (tcnghia) [reply README.md:5]: Done

File: README.md [hunk 3]
```diff
@@ -1,14 +1,54 @@
...
+* [http://github.com/knative-sandbox/net-istio] For the Istio-based
+  implementation of KIngress.
+
+# Knative Certificate aka KCert
+
+Knative Certificate CRD is a Knative abstraction for various SSL certificate
+provisioning solutions (such as cert-manager or self-signed SSL certificate).
+
+Check out:
+* pkg/apis/networking/v1alpha1/certificate_types.go for more information about
```
Comment (mattmoor) [inline README.md:49]: these too 😇

Comment (tcnghia) [reply README.md]: Done

File: README.md [hunk 4]
```diff
@@ -1,14 +1,54 @@
...
+* [http://github.com/knative-sandbox/net-kourier] for a dependency-free
+  implementation of KIngress based on Envoy proxy.
+* [http://github.com/knative-sandbox/net-istio] For the Istio-based
+  implementation of KIngress.
+
+# Knative Certificate aka KCert
+
+Knative Certificate CRD is a Knative abstraction for various SSL certificate
+provisioning solutions (such as cert-manager or self-signed SSL certificate).
+X
```
Comment (ZhiminXiang) [inline README.md:47]: nit: I think "X" is a typo here :)

File: README.md [hunk 5]
```diff
@@ -1,14 +1,52 @@
...
-[![Go Report Card](https://goreportcard.com/badge/knative/sample-controller)](https://goreportcard.com/report/knative/sample-controller)
+This repository contains the Knative Ingress and Certificate CRDs, as well as
+their conformance tests. These are our extension points to plugin different
+Ingress plugins (Ambassador, Contour, Gloo, Istio, Kong and Kourier), as well as
+different AutoTLS plugins (CertManager and Knative's own HTTP01 challenge
+solver).
 
-Knative `sample-controller` defines a few simple resources that are validated by
-webhook and managed by a controller to demonstrate the canonical style in which
-Knative writes controllers.
```
Comment (mattmoor) [inline README.md:9]: Format markdown:
```suggestion

```

File: README.md [hunk 6]
```diff
@@ -1,14 +1,52 @@
...
 
-To learn more about Knative, please visit our
-[Knative docs](https://github.com/knative/docs) repository.
+# Knative Ingress aka KIngress
 
-If you are interested in contributing, see [CONTRIBUTING.md](./CONTRIBUTING.md)
-and [DEVELOPMENT.md](./DEVELOPMENT.md).
+The Knative Ingress CRD is based on the Kubernetes Ingress CRD, with the
+following additions:
```
Comment (mattmoor) [inline README.md:13]: Format markdown:
```suggestion
following additions:

```

File: README.md [hunk 7]
```diff
@@ -1,14 +1,52 @@
...
+
+In addition to these, we previously added Timeout and Retry settings but no
+longer used them and may deprecate these parts in future versions.
+
+See
+https://knative.dev/docs/install/any-kubernetes-cluster/#installing-the-serving-component
+for a list of the current supported KIngress implementations.
+
+Check out:
+* [/pkg/apis/networking/v1alpha1/ingress_types.go](/pkg/apis/networking/v1alpha1/ingress_types.go)
```
Comment (mattmoor) [inline README.md:28]: Format markdown:
```suggestion

- [/pkg/apis/networking/v1alpha1/ingress_types.go](/pkg/apis/networking/v1alpha1/ingress_types.go)
```

File: README.md [hunk 8]
```diff
@@ -1,14 +1,52 @@
...
+  for more information about the KIngress API spec.
+* [/pkg/apis/networking/v1alpha1/ingress_validation.go](/pkg/apis/networking/v1alpha1/ingress_validations.go)
```
Comment (mattmoor) [inline README.md:30]: Format markdown:
```suggestion
- [/pkg/apis/networking/v1alpha1/ingress_validation.go](/pkg/apis/networking/v1alpha1/ingress_validations.go)
```

File: README.md [hunk 9]
```diff
@@ -1,14 +1,52 @@
...
+  for more information about the validation logic for KIngress API spec.
+* [/test/conformance/ingress/README.md](/test/conformance/ingress/README.md) for
```
Comment (mattmoor) [inline README.md:32]: Format markdown:
```suggestion
- [/test/conformance/ingress/README.md](/test/conformance/ingress/README.md) for
```

File: README.md [hunk 10]
```diff
@@ -1,14 +1,52 @@
...
+  the conformance tests and how to run them.
+
+See also:
+* [http://github.com/knative-sandbox/net-contour] for the Contour-based
```
Comment (mattmoor) [inline README.md:36]: Format markdown:
```suggestion

- [http://github.com/knative-sandbox/net-contour] for the Contour-based
```

File: README.md [hunk 11]
```diff
@@ -1,14 +1,52 @@
...
+  implementation of KIngress.
+* [http://github.com/knative-sandbox/net-kourier] for a dependency-free
```
Comment (mattmoor) [inline README.md:38]: Format markdown:
```suggestion
- [http://github.com/knative-sandbox/net-kourier] for a dependency-free
```

File: README.md [hunk 12]
```diff
@@ -1,14 +1,52 @@
...
+  implementation of KIngress based on Envoy proxy.
+* [http://github.com/knative-sandbox/net-istio] For the Istio-based
```
Comment (mattmoor) [inline README.md:40]: Format markdown:
```suggestion
- [http://github.com/knative-sandbox/net-istio] For the Istio-based
```

File: README.md [hunk 13]
```diff
@@ -1,14 +1,52 @@
...
+  implementation of KIngress.
+
+# Knative Certificate aka KCert
+
+Knative Certificate CRD is a Knative abstraction for various SSL certificate
+provisioning solutions (such as cert-manager or self-signed SSL certificate).
+
+Check out:
+* [/pkg/apis/networking/v1alpha1/certificate_types.go](/pkg/apis/networking/v1alpha1/certificate_types.go)
```
Comment (mattmoor) [inline README.md:49]: Format markdown:
```suggestion

- [/pkg/apis/networking/v1alpha1/certificate_types.go](/pkg/apis/networking/v1alpha1/certificate_types.go)
```

File: README.md [hunk 14]
```diff
@@ -1,14 +1,52 @@
...
+  for more information about the Certificate API spec.
+* [/pkg/apis/networking/v1alpha1/certificate_validation.go](/pkg/apis/networking/v1alpha1/certificate_validations.go)
```
Comment (mattmoor) [inline README.md:51]: Format markdown:
```suggestion
- [/pkg/apis/networking/v1alpha1/certificate_validation.go](/pkg/apis/networking/v1alpha1/certificate_validations.go)
```

File: README.md [hunk 15]
```diff
@@ -1,14 +1,52 @@
...
 
-Knative `sample-controller` defines a few simple resources that are validated by
-webhook and managed by a controller to demonstrate the canonical style in which
-Knative writes controllers.
+# Knative Ingress aka KIngress
 
-To learn more about Knative, please visit our
-[Knative docs](https://github.com/knative/docs) repository.
+The Knative Ingress CRD is based on the Kubernetes Ingress CRD, with the
+following additions:
```
Comment (mattmoor) [inline README.md:12]: Format markdown:
```suggestion
following additions:

```

File: README.md [hunk 16]
```diff
@@ -1,14 +1,52 @@
...
+longer used them and may deprecate these parts in future versions.
+
+See
+https://knative.dev/docs/install/any-kubernetes-cluster/#installing-the-serving-component
+for a list of the current supported KIngress implementations.
+
+Check out:
+
+- [/pkg/apis/networking/v1alpha1/ingress_types.go](/pkg/apis/networking/v1alpha1/ingress_types.go)
+  for more information about the KIngress API spec.
```
Comment (mattmoor) [inline README.md:29]: Format markdown:
```suggestion
  for more information about the KIngress API spec.

```

File: README.md [hunk 17]
```diff
@@ -1,14 +1,52 @@
...
+* [/pkg/apis/networking/v1alpha1/ingress_validation.go](/pkg/apis/networking/v1alpha1/ingress_validations.go)
+  for more information about the validation logic for KIngress API spec.
+* [/test/conformance/ingress/README.md](/test/conformance/ingress/README.md) for
+  the conformance tests and how to run them.
+
+See also:
+* [http://github.com/knative-sandbox/net-contour] for the Contour-based
```
Comment (mattmoor) [inline README.md:36]: Format markdown:
```suggestion

- [http://github.com/knative-sandbox/net-contour] for the Contour-based
```

File: README.md [hunk 18]
```diff
@@ -1,14 +1,52 @@
...
+  implementation of KIngress.
+* [http://github.com/knative-sandbox/net-kourier] for a dependency-free
```
Comment (mattmoor) [inline README.md:38]: Format markdown:
```suggestion
- [http://github.com/knative-sandbox/net-kourier] for a dependency-free
```

File: README.md [hunk 19]
```diff
@@ -1,14 +1,52 @@
...
+  implementation of KIngress based on Envoy proxy.
+* [http://github.com/knative-sandbox/net-istio] For the Istio-based
```
Comment (mattmoor) [inline README.md:40]: Format markdown:
```suggestion
- [http://github.com/knative-sandbox/net-istio] For the Istio-based
```

File: README.md [hunk 20]
```diff
@@ -1,14 +1,52 @@
...
+  implementation of KIngress.
+
+# Knative Certificate aka KCert
+
+Knative Certificate CRD is a Knative abstraction for various SSL certificate
+provisioning solutions (such as cert-manager or self-signed SSL certificate).
+
+Check out:
+* [/pkg/apis/networking/v1alpha1/certificate_types.go](/pkg/apis/networking/v1alpha1/certificate_types.go)
```
Comment (mattmoor) [inline README.md:49]: Format markdown:
```suggestion

- [/pkg/apis/networking/v1alpha1/certificate_types.go](/pkg/apis/networking/v1alpha1/certificate_types.go)
```

File: README.md [hunk 21]
```diff
@@ -1,14 +1,52 @@
...
+  for more information about the Certificate API spec.
+* [/pkg/apis/networking/v1alpha1/certificate_validation.go](/pkg/apis/networking/v1alpha1/certificate_validations.go)
```
Comment (mattmoor) [inline README.md:51]: Format markdown:
```suggestion
- [/pkg/apis/networking/v1alpha1/certificate_validation.go](/pkg/apis/networking/v1alpha1/certificate_validations.go)
```

File: README.md [hunk 22]
```diff
@@ -1,14 +1,54 @@
...
+- [/pkg/apis/networking/v1alpha1/ingress_types.go](/pkg/apis/networking/v1alpha1/ingress_types.go)
+  for more information about the KIngress API spec.
+
+* [/pkg/apis/networking/v1alpha1/ingress_validation.go](/pkg/apis/networking/v1alpha1/ingress_validations.go)
+  for more information about the validation logic for KIngress API spec.
+* [/test/conformance/ingress/README.md](/test/conformance/ingress/README.md) for
+  the conformance tests and how to run them.
+
+See also:
+* [http://github.com/knative-sandbox/net-contour] for the Contour-based
```
Comment (mattmoor) [inline README.md:38]: Format markdown:
```suggestion

- [http://github.com/knative-sandbox/net-contour] for the Contour-based
```

File: README.md [hunk 23]
```diff
@@ -1,14 +1,54 @@
...
+  implementation of KIngress.
+- [http://github.com/knative-sandbox/net-kourier] for a dependency-free
```
Comment (mattmoor) [inline README.md:40]: Format markdown:
```suggestion
- [http://github.com/knative-sandbox/net-kourier] for a dependency-free
```

File: README.md [hunk 24]
```diff
@@ -1,14 +1,54 @@
...
+  implementation of KIngress based on Envoy proxy.
+* [http://github.com/knative-sandbox/net-istio] For the Istio-based
```
Comment (mattmoor) [inline README.md:42]: Format markdown:
```suggestion
- [http://github.com/knative-sandbox/net-istio] For the Istio-based
```

File: README.md [hunk 25]
```diff
@@ -1,14 +1,54 @@
...
+  implementation of KIngress.
+
+# Knative Certificate aka KCert
+
+Knative Certificate CRD is a Knative abstraction for various SSL certificate
+provisioning solutions (such as cert-manager or self-signed SSL certificate).
+
+Check out:
+* [/pkg/apis/networking/v1alpha1/certificate_types.go](/pkg/apis/networking/v1alpha1/certificate_types.go)
```
Comment (mattmoor) [inline README.md:51]: Format markdown:
```suggestion

- [/pkg/apis/networking/v1alpha1/certificate_types.go](/pkg/apis/networking/v1alpha1/certificate_types.go)
```

File: README.md [hunk 26]
```diff
@@ -1,14 +1,54 @@
...
+  for more information about the Certificate API spec.
+* [/pkg/apis/networking/v1alpha1/certificate_validation.go](/pkg/apis/networking/v1alpha1/certificate_validations.go)
```
Comment (mattmoor) [inline README.md:53]: Format markdown:
```suggestion
- [/pkg/apis/networking/v1alpha1/certificate_validation.go](/pkg/apis/networking/v1alpha1/certificate_validations.go)
```

File: README.md (hunk 22 above)
Comment (mattmoor) [inline README.md:38]: Format markdown:
```suggestion

- [http://github.com/knative-sandbox/net-contour] for the Contour-based
```

File: README.md (hunk 23 above)
Comment (mattmoor) [inline README.md:40]: Format markdown:
```suggestion

* [http://github.com/knative-sandbox/net-kourier] for a dependency-free
```

File: README.md (hunk 24 above)
Comment (mattmoor) [inline README.md:42]: Format markdown:
```suggestion

- [http://github.com/knative-sandbox/net-istio] For the Istio-based
```

File: README.md (hunk 25 above)
Comment (mattmoor) [inline README.md:51]: Format markdown:
```suggestion

- [/pkg/apis/networking/v1alpha1/certificate_types.go](/pkg/apis/networking/v1alpha1/certificate_types.go)
```

File: README.md (hunk 26 above)
Comment (mattmoor) [inline README.md:53]: Format markdown:
```suggestion
- [/pkg/apis/networking/v1alpha1/certificate_validation.go](/pkg/apis/networking/v1alpha1/certificate_validations.go)
```

File: README.md (hunk 4 above)
Comment (tcnghia) [reply README.md]: Good catch. Done now.

File: README.md [hunk 27]
```diff
@@ -1,14 +1,56 @@
...
+- [http://github.com/knative-sandbox/net-istio] For the Istio-based
+  implementation of KIngress.
+
+# Knative Certificate aka KCert
+
+Knative Certificate CRD is a Knative abstraction for various SSL certificate
+provisioning solutions (such as cert-manager or self-signed SSL certificate).
+
+Check out:
+* [/pkg/apis/networking/v1alpha1/certificate_types.go](/pkg/apis/networking/v1alpha1/certificate_types.go)
```
Comment (mattmoor) [inline README.md:53]: Format markdown:
```suggestion

- [/pkg/apis/networking/v1alpha1/certificate_types.go](/pkg/apis/networking/v1alpha1/certificate_types.go)
```

File: README.md [hunk 28]
```diff
@@ -1,14 +1,56 @@
...
+  for more information about the Certificate API spec.
+- [/pkg/apis/networking/v1alpha1/certificate_validation.go](/pkg/apis/networking/v1alpha1/certificate_validations.go)
```
Comment (mattmoor) [inline README.md:55]: Format markdown:
```suggestion
- [/pkg/apis/networking/v1alpha1/certificate_validation.go](/pkg/apis/networking/v1alpha1/certificate_validations.go)
```

File: README.md [hunk 29]
```diff
@@ -1,14 +1,52 @@
...
+- [/test/conformance/ingress/README.md](/test/conformance/ingress/README.md) for
+  the conformance tests and how to run them.
+
+See also:
+- [http://github.com/knative-sandbox/net-contour] for the Contour-based
+  implementation of KIngress.
+- [http://github.com/knative-sandbox/net-kourier] for a dependency-free
+  implementation of KIngress based on Envoy proxy.
+- [http://github.com/knative-sandbox/net-istio] For the Istio-based
+  implementation of KIngress.
```
Comment (mattmoor) [inline README.md:41]: I don't think you need the brackets on these, the rendered view looks funny.

File: README.md [hunk 30]
```diff
@@ -1,14 +1,52 @@
...
+See
+https://knative.dev/docs/install/any-kubernetes-cluster/#installing-the-serving-component
+for a list of the current supported KIngress implementations.
+
+Check out:
+- [/pkg/apis/networking/v1alpha1/ingress_types.go](/pkg/apis/networking/v1alpha1/ingress_types.go)
+  for more information about the KIngress API spec.
+- [/pkg/apis/networking/v1alpha1/ingress_validation.go](/pkg/apis/networking/v1alpha1/ingress_validations.go)
+  for more information about the validation logic for KIngress API spec.
+- [/test/conformance/ingress/README.md](/test/conformance/ingress/README.md) for
```
Comment (mattmoor) [inline README.md:32]: maybe drop the leading `/` or maybe even the path (outside of the link) to clean this up?  Same below

**Pull request merged**